integration_tests/tests/s3.rs

```diff
@@ -461,3 +469,49 @@ fn test_get_object_with_metadata(client: &TestClient, bucket: &str, filename: &s
     let head_metadata = result.metadata.as_ref().expect("No metadata available");
     assert_eq!(metadata, head_metadata);
 }
+
+fn test_generate_presigned_url(region: &Region, credentials: &AwsCredentials, bucket: &str, filename: &str) {
+    use reqwest::header::ContentLength;
+    use rusoto_s3::util::PreSignedRequest;
+    fn check(mut res: reqwest::Response){
+        assert!(&res.status().is_success());
+        let size = res.headers().get::<ContentLength>().map(|ct_len| **ct_len).unwrap_or(0);
+        assert!(size > 0);
+        let mut buf: Vec<u8> = vec![];
+        res.copy_to(&mut buf).unwrap();
+        assert!(buf.len() > 0);
+    }
+    {
+        let req = GetObjectRequest {
+            bucket: bucket.to_owned(),
+            key: filename.to_owned(),
+            ..Default::default()
+        };
+        let presigned_url = req.get_presigned_url(region, credentials);
+        println!("get object presigned url: {:#?}", presigned_url);
+        let res = reqwest::get(&presigned_url).unwrap();
```

Comment:
```
get object presigned url: "http://localhost:9000/rusoto-test-bucket-1523876092/test_file_1523876092?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ANTN35UAENTS5UIAEATD%2F20180416%2Fus-east-1%20with%20endpoint%20http%3A%2F%2Flocalhost%3A9000%2Fs3%2Faws4_request&X-Amz-Date=20180416T105458Z&X-Amz-Expires=3600&X-Amz-Signature=0043a9e9d5759e04b058e61f3cb8bb99c6aa666bb922cb81b04ca7141b26250f&X-Amz-SignedHeaders=content-type%3Bhost%3Bx-amz-content-sha256%3Bx-amz-date&x-amz-content-sha256=&x-amz-date=20180416T105458Z"
thread 'test_all_the_things' panicked at 'assertion failed: &res.status().is_success()', tests/s3.rs:477:9
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at libstd/sys_common/backtrace.rs:59
             at libstd/panicking.rs:380
   3: std::panicking::default_hook
             at libstd/panicking.rs:390
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:576
   5: std::panicking::begin_panic
   6: s3::test_generate_presigned_url::check
   7: s3::test_generate_presigned_url
   8: s3::test_all_the_things
   9: <F as alloc::boxed::FnBox<A>>::call_box
             at libtest/lib.rs:1462
             at /checkout/src/libcore/ops/function.rs:223
             at /checkout/src/liballoc/boxed.rs:788
  10: __rust_maybe_catch_panic
             at libpanic_unwind/lib.rs:102


failures:
    test_all_the_things

test result: FAILEDB. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '--test s3'
ERROR: Docker test failed: image 'ceph/demo', container 'd466d8aeddb6'
terminating docker: image 'ceph/demo', container 'd466d8aeddb6'
d466d8aeddb645c1248465b10078a907acf7cc2e95dd828dbdfb87bf5f885ba6
run-parts: .semaphoreci/test_10_ceph.sh exited with return code 1
```

This test is still failed. 
Is it able to create presigned url on localhost server?

Reply:
Ceph _should_ support pre-signed URLs from what I can gather. Could this be a bug in the pre-signed url generation code?

Reply:
Take a look at https://tracker.ceph.com/issues/16463, could this be related?